server/svix-server/src/error.rs

```diff
@@ -16,35 +16,53 @@ use serde_json::json;
 pub type Result<T> = std::result::Result<T, Error>;
 
 /// The error type returned from the Svix API
-#[derive(Debug, Clone)]
-pub enum Error {
-    /// A generic error
-    Generic(String),
-    /// Database error
-    Database(String),
-    /// Queue error
-    Queue(String),
-    /// Database error
-    Validation(String),
-    /// Any kind of HttpError
-    Http(HttpError),
+#[derive(Debug)]
+pub struct Error {
+    // the file name and line number of the error. Used for debugging non Http errors
+    pub location: &'static str,
```

Comment:
Correct me if I'm wrong, but we are not maintaining the error chain? E.g. we are missing location (and type) of the caller? Maybe it makes sense to have a linked-list type of thing of vectors (or heck, a pre-allocated vec of size 10 that should always be sufficient) so we keep the trace?

Reply:
Every time we call `ctx!` btw, obviously it'll be lost if we don't.

Reply:
Yeah, good catch.

Should be updated now so that `ctx!` also adds the tracing (I ended up renaming the `IntoSvixError` trait, hopefully it makes a bit more sense now).

PR description should also be updated showing what the end "trace" result looks like.

I didn't go through and add `ctx!(...)` to _every_ single relevant callsite, though I can if you want (it'll be a bit tedious). Unfortunately, I'm not sure how to make forgetting to call `ctx!` on a `crate::error::Result` a compiler error like I was when forgetting to call `ctx!` on a normal `Result`.

Reply:
Nah, no need to add it everywhere for now.

Btw, I remember you preferred `e!`, why the change? (I don't care much either way)

Reply:
Haha, I still _personally_ prefer `e!` just for brevity. But I also think `ctx!` is more descriptive / it's easier to infer what `ctx!` is doing.

Maybe not though 🤷.

Reply:
I think I agree. :)